### PR TITLE
Update LuaJIT

### PR DESCRIPTION
**Pull request by nuoun**
Initial test PR

- LuaJIT submodule now pointing to v2.1 branch
- Updated custom msvcbuild.bat
- Some text edits in README.md and code comments

### CODE REVIEW

**Comment by baconpaul** [comment]
Looks good. Ready to merge?

**Comment by nuoun** [comment]
So the msvc built lua51.lib went from 3.5mb to almost 10mb and I'm not sure why, I'm pretty sure it's not a debug build.

File: libs/luajitlib/msvcbuild.bat [hunk 1]
```diff
@@ -13,9 +13,15 @@
...
 
 @setlocal
-@rem Add more debug flags here, e.g. DEBUGCFLAGS=/DLUA_USE_APICHECK
+@rem Add more debug flags here, e.g. DEBUGCFLAGS=/DLUA_USE_ASSERT
 @set DEBUGCFLAGS=
 @set LJCOMPILE=cl /nologo /c /O2 /W3 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_STDIO_INLINE=__declspec(dllexport)__inline
+@set LJDYNBUILD=/DLUA_BUILD_AS_DLL /MT
+@set LJDYNBUILD_DEBUG=/DLUA_BUILD_AS_DLL /MTd 
+@set LJCOMPILETARGET=/Zi
+@set LJLINKTYPE=/DEBUG /RELEASE
```
Comment (baconpaul) [inline libs/luajitlib/msvcbuild.bat:22]: Could this line be why? Why does it say debug release?

**Comment by baconpaul** [comment]
Flagged one line in the diff which maybe is it?

**Comment by nuoun** [comment]
Neither changing 
LJDYNBUILD_DEBUG to =/DLUA_BUILD_AS_DLL /MT
or LJLINKTYPE to just =/RELEASE
seems to make a difference, the .a files also got a bit bigger but this seems just bloated?

**Comment by nuoun** [comment]
"/Zi does imply /debug" apparently

**Comment by nuoun** [comment]
Ok ready to merge if this passes

**Comment by baconpaul** [comment]
Ok!

**Pull request merged**